backend/scripts/github_collector.py

```
@@ -50,23 +103,113 @@ def _handle_rate_limit(self, e: RateLimitExceededException):
+            
+            has_updates = stored_sha != current_sha
+            
+            return {
+                "update_available": has_updates, # Normalized key for frontend
+                "has_updates": has_updates,      # Legacy key
+                "local_commit": stored_sha,
+                "remote_commit": current_sha,
+                "local_date": local_metadata.get("latest_commit_date"),
+                "remote_date": latest_commit.commit.author.date.isoformat() if latest_commit.commit.author else None
+            }
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)